lib/e3db/client.rb

```diff
@@ -559,24 +621,226 @@ def revoke(type, reader_id)
       @conn.put(url, JSON.generate({:deny => [{:read => {}}]}))
 
       delete_access_key(id, id, reader_id, type)
+      nil
     end
 
+    # Gets a list of record types that this client has shared with
+    # others.
+    #
+    # @return [Array<OutgoingSharingPolicy>]
     def outgoing_sharing
       url = get_url('v1', 'storage', 'policy', 'outgoing')
       resp = @conn.get(url)
       json = JSON.parse(resp.body, symbolize_names: true)
       return json.map {|x| OutgoingSharingPolicy.new(x)}
     end
 
+    # Gets a list of record types that others have shared with this
+    # client.
+    #
+    # @return [Array<IncomingSharingPolicy>]
     def incoming_sharing
       url = get_url('v1', 'storage', 'policy', 'incoming')
       resp = @conn.get(url)
       json = JSON.parse(resp.body, symbolize_names: true)
       return json.map {|x| IncomingSharingPolicy.new(x)}
     end
 
+    # Create and return an (encrypted) secret key for encrypting the
+    # given record type. Can be saved for use with {#encrypt_record},
+    # {#encrypt_new_record} and {#decrypt_record} later.
+    #
+    # @return [EAK]
+    def create_writer_key(type)
+      id = @config.client_id
+      begin
+        put_access_key(id, id, id, type, new_access_key)
+      rescue Faraday::ClientError => e
+        # Ignore 403, as it means a key already exists. Otherwise, raise.
+        if e.response[:status] != 403
+          raise e
+        end
+      end
+
+      get_eak(id, id, type)
+    end
+
+    # Retrieve an (encrypted) key for reading records written by the
+    # given writer for the given user with the given type (assuming
+    # the writer previously established sharing with this client).
+    #
+    # @return [EAK]
+    def get_reader_key(writer_id, user_id, type)
```

Comment:
Not sure why we need this separate interface ... why not just call `get_eak()` since the arguments are the same for both?